File: src/main/java/com/digsol/turnomed/mapper/AddressesMapper.java
package com.digsol.turnomed.mapper;

import com.digsol.turnomed.model.Address;
import com.digsol.turnomed.model.Users;
import org.apache.ibatis.annotations.Mapper;
import org.springframework.stereotype.Component;


import java.util.List;

@Mapper
@Component
public interface AddressesMapper {
    int deleteByPrimaryKey(Integer id);

    int insert(Address record);

    int insertSelective(Address record);

    Address selectByPrimaryKey(Integer id);

    int updateByPrimaryKeySelective(Address record);

    int updateByPrimaryKey(Address record);

    List<Address> selectAll();

    Address getAddressInformation(Integer addressId);

    Address getAddressInformationByEmail(Users users);

   // List<EventsSchedule> detailEventsInfoByAddress(SearchEntity request);
}

File: src/main/kotlin/com/digsol/turnomed/mapper/AddressesMapper.kt
package com.digsol.turnomed.mapper

import com.digsol.turnomed.model.Address
import com.digsol.turnomed.model.Users
import org.apache.ibatis.annotations.Mapper
import org.springframework.stereotype.Component

@Mapper
@Component(value = "addressesMapper")
interface AddressesMapper {

    fun deleteByPrimaryKey(id: Integer): Int

    fun insert(record: Address): Int

    fun insertSelective(record: Address): Int

    fun selectByPrimaryKey(id: Integer): Address

    fun updateByPrimaryKeySelective(record: Address): Int

    fun updateByPrimaryKey(record: Address): Int

    fun selectAll(): List<Address>

    fun getAddressInformation(addressId: Integer): Address

    fun getAddressInformationByEmail(users: Users): Address

    fun detailEventsInfoByAddress(request: SearchEntity): List<EventsSchedule>
}
